website/project/model.py

```diff
@@ -790,6 +790,31 @@ class Node(GuidStoredObject, AddonModelMixin, IdentifierMixin):
 
     piwik_site_id = fields.StringField()
 
+    # Primary institution node is attached to
+    primary_institution = fields.ForeignField('institution', backref='node')
```

Comment:
Please remove this backref. `Node.find(Q('primary_institution', 'eq', institution._id))` would work as well when you need to find the associated nodes.